Source/DSPSpy/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+cmake_minimum_required(VERSION 3.5.0)
+
+project(dspspy)
+
+set(CMAKE_EXECUTABLE_SUFFIX ".elf")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+set(MACHDEP "-DHW_RVL -DRVL -DGEKKO -mrvl -mcpu=750 -meabi -mhard-float")
+set(CMAKE_C_FLAGS "${CMAKE_CXX_FLAGS} ${MACHDEP}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${MACHDEP}")
+
+include_directories(${LIBOGCDIR}/include)
+link_directories(${LIBOGCDIR}/lib/wii)
```

Comment:
This can be passed in the `target_link_libraries` command.

Reply:
Putting this lib directory in the `target_link_libraries` command doesn't seem to work. cmake says you can't link against a directory. I figured out that you *can* specify full paths to the library file (e.g. `${LIBOGCDIR}/lib/wii/libwiiuse.a`), but that seems pretty ugly to me. What would be the preferred approach?

Reply:
Hmm, apparently modern cmake would have you use `add_library(name ... IMPORTED)`, etc. which seems like a lot of work just to link against libogc standard libraries... I'd love to hear your opinion on this though.

Reply:
Right, whoops. I overlooked that target_link_libraries expects absolute individual paths when specifying libs directly and not by target.

Since it's a truly external library (i.e. not inside the externals folder, which is the case here), it's likely better to write a Findxxx.cmake module file like what's in the CMake folder (since the library can potentially not even be installed on a person's system). Mainly because you can use `find_package` to hard-fail if it can't be found. 

e.g. Say a hypothetical FindLibOGC.cmake file defined the target `devkitpro::libogc`. It would be utilized like so:

```cmake
...

find_package(LibOGC REQUIRED)

...

target_link_libraries(your_target
    PRIVATE
        devkitpro::libogc
)
```

That's what I would try and do first personally anyway. Since it keeps the path config and setup centralized in its own module file separate from other targets.

Though this is probably a question @Orphis is more suited to answer.

Reply:
Ah, I see. If you don't mind, I have an additional question (for you or Orphis): how would this work with optional modules? For example, if the Wiimote stuff isn't needed (for a homebrew with no video output), would it be possible when using find_package(LibOGC) to skip linking against unused libraries, or would there need to be several packages/targets, e.g. one for libogc core, one for Wiimotes, etc.?

Reply:
No `link_directories()` call allowed, just link the actual library or make an imported target for it and link against it.
For all that, it would be better to make a toolchain file for the GC toolchain directly and add this library since it's always required. It has nothing to do in the dspspy build scripts also.